src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,199 @@
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("all_proxy");
+            if (value != null)
+            {
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("http_proxy");
+            if (value != null)
+            {
+                http = value;
+            }
+            value = Environment.GetEnvironmentVariable("HTTPS_PROXY");
+            if (value != null)
+            {
+                https = value;
+            }
+
+            value = Environment.GetEnvironmentVariable("https_proxy");
+            if (value != null)
+            {
+                https = value;
+            }
+            // fail to instantiate if nothing is set.
+            // caller may pick some other proxy type.
+            if (String.IsNullOrWhiteSpace(http) && String.IsNullOrWhiteSpace(https))
+            {
+                return null;
+            }
+
+            return new HttpEnvironmentProxy(http, https);
+        }
+
+        private HttpEnvironmentProxy(string http, string https)
+        {
+
+            if (!String.IsNullOrWhiteSpace(http))
+            {
+                _http = GetUriFromString(http);
+                if (!string.IsNullOrWhiteSpace(_http.UserInfo))
+                {
+                    String[] s = _http.UserInfo.Split(':', 2);
+                    _httpCred = new NetworkCredential(s[0], s[1]);
+                }
+            }
+            if (!String.IsNullOrWhiteSpace(https))
+            {
+                _https = GetUriFromString(https);
+                if (!string.IsNullOrWhiteSpace(_https.UserInfo))
+                {
+                    String[] s = _https.UserInfo.Split(':', 2);
+                    _httpsCred = new NetworkCredential(s[0], s[1]);
+                }
+            }
+
+            string value = Environment.GetEnvironmentVariable("no_proxy");
+            if (!String.IsNullOrWhiteSpace(value)) 
+            {
+                _bypass = value.Split(',');
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. 
+        /// </summary>
+        private Uri GetUriFromString(String value)
+        {
+            if (!value.StartsWith("http://", StringComparison.OrdinalIgnoreCase) &&
+                !value.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+            {
+                // Default to HTTP
+                value = "http://" + value;
+            }
+            Uri uri = new Uri(value);
+
+            if (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps)
+            {
+                throw new NotSupportedException("Only HTTP and HTTPS protocols are supported for proxy.");
```

Comment:
We need to consider what behavior we want on error here.  As written, this will cause every request to throw.  Is that what we want?  Alternatively, we could silently discard errors, or at least certain ones.  I'm inclined to eat the errors unless we really cannot proceed at all.  Thoughts?

Reply:
Fine with me. I think there are possibly two cases. Value libcurl considers as valid but we don't - like sock/socks5. The other is completely bogus value. I did some testing and libcurl seems to simply ignore it e.g. does what you suggested.